client/functions.lua

```diff
@@ -865,18 +857,22 @@ function QBCore.Functions.SetVehicleProperties(vehicle, props)
 end
 
 function QBCore.Functions.LoadParticleDictionary(dictionary)
-    if not HasNamedPtfxAssetLoaded(dictionary) then
-        RequestNamedPtfxAsset(dictionary)
-        while not HasNamedPtfxAssetLoaded(dictionary) do
-            Wait(0)
-        end
+    if HasNamedPtfxAssetLoaded(dictionary) then return end
+    RequestNamedPtfxAsset(dictionary)
+    while not HasNamedPtfxAssetLoaded(dictionary) do
+        Wait(0)
     end
 end
 
-function QBCore.Functions.StartParticleAtCoord(Dict, ptName, looped, coords, rot, scale, alpha, color, duration)
-    QBCore.Functions.LoadParticleDictionary(Dict)
-    UseParticleFxAssetNextCall(Dict)
-    SetPtfxAssetNextCall(Dict)
+function QBCore.Functions.StartParticleAtCoord(dict, ptName, looped, coords, rot, scale, alpha, color, duration)
+    if coords then
+        coords = type(coords) == 'table' and vec3(coords.x, coords.y, coords.z) or coords
+    else
+        coords = GetEntityCoords(ped)
```

Comment:
```suggestion
        coords = GetEntityCoords(PlayerPedId())
```
Line 405 needs to be fixed with this as well.